src/backend/utils/adt/agtype.c

```diff
@@ -2788,6 +2789,9 @@ Datum agtype_to_float8(PG_FUNCTION_ARGS)
     else if (agtv.type == AGTV_NUMERIC)
         result = DatumGetFloat8(DirectFunctionCall1(numeric_float8,
                      NumericGetDatum(agtv.val.numeric)));
+    else if (agtv.type == AGTV_STRING)
+        result = DatumGetFloat8(DirectFunctionCall1(float8in,
+                                                    CStringGetDatum(agtv.val.string.val)));
     else
```

Comment:
When we work on functions that don't conform to our standards, we proactively try to clean them up. Just the functions that we are working on. If you could adjust the function to have {} where necessary.